File: 0191-number-of-1-bits/0191-number-of-1-bits.go
/*
func hammingWeight(num uint32) int {
    if num == 0 {
        return 0
    } 
    
    count := 0
    for num > 1 {
        if num % 2 == 1 {
            count++
        }
        num >>= 1
    }
    
    return count + 1
}
*/

func hammingWeight(num uint32) int {
    if num == 0 {
        return 0
    } 
    
    count := 0
    for num > 1 {
        if num % 2 == 1 {
            count++
        }
        num /= 2
    }
    
    return count + 1
}

File: 0191-number-of-1-bits/0191-number-of-1-bits.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        if n == 0:
            return 0
        
        count = 0
        while n > 1:
            if n % 2 == 1:
                count += 1
            n = n >> 1
            
        return count + 1  